### PR TITLE
Segementiation overlay

### PR DESCRIPTION
**Pull request by cleaaum**
**Description**: This PR adds a toggle in the Model Configuration tab to allow user to view segmented results overlayed on top of the initial dataset. There is an opacity slider to adjust the visibility of the segmented result. 

**Logic**: the segmentation results image gets rendered in the `render_image()` callback (this takes a bit of time). Once the image is rendered, we can user `Patch()` to hide and show this image a little faster when the toggle is used, and we can  adjust the opacity of the results with the slider in the same way. This happens in the `hide_show_segmentation_overlay()` callback.

**Note**: the color scale for the overlayed image will be updated in a future PR (when we have more seg images available for testing).

### CODE REVIEW

**Comment by hannahker** [comment]
Adding a screenshot of the UI for reference:
Screenshot 2023-09-19 at 8 54 34 AM

**Comment by hannahker** [comment]
Great start on this! We'll have to refine some things further on this feature but I think this is almost ready to merge. See #121 for some minor tweaks I suggested. Could we also toggle the results off by default when someone loads the page?

**Comment by cleaaum** [comment]
@hannahker Ive added that default behaviour where the seg results are switched off by default and i've approved you #121 ! 💃

File: callbacks/control_bar.py [hunk 1]
```diff
@@ -800,3 +801,47 @@ def open_controls_drawer(n_clicks, is_opened):
...
+        disabled_dropdown = dropdown_enabled
+        disabled_toggle = False
+        disabled_slider = slider_enabled
+    else:
+        results = [
+            item
+            for item in get_data_project_names()
+            if ("seg" in item and image_src in item)
+        ]
+        if results:
```
Comment (Wiebke) [inline callbacks/control_bar.py:840]: For some reason, this does not yield a populated dropdown when I am testing against the Tiled instance on Spin. 

However, I am interestingly now also hitting the timeout problems that you have been seeing before.

Those don't show up in a fully reproducible way, but I seem to hit them most consistently when I switch projects. 

I wonder if the timeout for retrieving data, also affects this callback, then causing for the segmentation result not to show up correctly.

**Review by Wiebke — COMMENTED** [review]
Testing locally works and looks great.

For testing with the Spin instance I do now occasionally run into the timeouts issue as well.

File: callbacks/control_bar.py (hunk 1 above)
Comment (Wiebke) [reply callbacks/control_bar.py:833]: An update on this: The timeout was indeed the issue. With no time-outs the dropdown gets correctly populated and overlaying works great too.

**Comment by cleaaum** [comment]
This PR now works with the project names hardocoded instead of using the `get_data_project_names()` function and it seems to run perfectly fine without any timeouts. I believe that function was causing too many requests to the Tiled server. I think we can perhaps leave them hardcoded for now and update this when caching is implemented. cc @hannahker

**Comment by hannahker** [comment]
Ok hopefully final item 😄 I tested this out on `staging` and the overlay looks great! I'll note that it _does_ definitely introduce more of a lag in the image rendering and display, but this is to be expected with bringing in more data. Just something to be aware of and it might be appropriate to add in an option to downsample the results in future work. 

The only bug I can find at this stage is in changing modes in the drawing toolbar. This doesn't seem to be working, either through a click or with a keybind. Do you know why this might be?

**Comment by cleaaum** [comment]
Oh my!! ahahah that's annoying.. and yes I know why... I specifically wanted to address the fact that the loading was slow by adding a loading overlay when you toggle the seg results on. I adjusted the following callback, and for some reason it prevents you from switching the annotation tool 😕 ? If you remove the last 2 inputs, it works just fine...  I have no idea as to why that would happen since we're just updating the zindex..

```
clientside_callback(
    """
    function EnableImageLoadingOverlay(zIndexSlider,zIndexToggle,zIndexSelector) {
        return 9999;
    }
    """,
    Output("image-viewer-loading", "zIndex"),
    Input("image-selection-slider", "value"),
    Input("show-result-overlay-toggle", "checked"),
    Input("annotated-slices-selector", "value"),
)
```

**Comment by cleaaum** [comment]
Ok so the last input of that callback was causing that bug - this was to add a loading overlay when the user jumps to a new slice by using the slice selection dropdown. We can adjust that in a different PR. The bug is gone now.. @hannahker

**Comment by hannahker** [comment]
Ok mostly gone but I'm still finding the very specific case where I can't adjust the drawing mode after I jump to a slice using the "Annotated slices" dropdown. Do you see this on your end as well?

**Review by hannahker — APPROVED** [review]

**Comment by hannahker** [comment]
Approving now! Discussed with @cleaaum and this bug is already on the `main` branch and can be fixed separately.

**Pull request merged**